ios/RCTContacts/RCTContacts.m

```diff
@@ -61,41 +61,132 @@ - (NSDictionary *)constantsToExport
+    [self getContactsFromAddressBook:contactStore matchingString:string withThumbnails:withThumbnails callback:callback];
 }
 
 -(void) getContactsFromAddressBook:(CNContactStore *)store
                     matchingString:(NSString *)searchString
+                    withThumbnails:(BOOL) withThumbnails
                           callback:(RCTResponseSenderBlock)callback
 {
-    NSMutableArray *contacts = [[NSMutableArray alloc] init];
+    NSMutableArray<CNContact*> *contacts = [[NSMutableArray alloc] init];
     NSError *contactError = nil;
-    NSArray *keys = @[
-                      CNContactEmailAddressesKey,
-                      CNContactPhoneNumbersKey,
-                      CNContactFamilyNameKey,
-                      CNContactGivenNameKey,
-                      CNContactMiddleNameKey,
-                      CNContactPostalAddressesKey,
-                      CNContactOrganizationNameKey,
-                      CNContactJobTitleKey,
-                      CNContactImageDataAvailableKey,
-                      CNContactNoteKey,
-                      CNContactUrlAddressesKey,
-                      CNContactBirthdayKey
-                      ];
-    NSArray *arrayOfContacts = [store unifiedContactsMatchingPredicate:[CNContact predicateForContactsMatchingName:searchString]
-                                                           keysToFetch:keys
-                                                                 error:&contactError];
-    [arrayOfContacts enumerateObjectsUsingBlock:^(CNContact * _Nonnull obj, NSUInteger idx, BOOL * _Nonnull stop) {
-        NSDictionary *contactDictionary = [self contactToDictionary:obj withThumbnails:NO];
-        [contacts addObject:contactDictionary];
+
+    NSArray *nameKeys = @[
+        CNContactFamilyNameKey,
+        CNContactGivenNameKey,
+        CNContactMiddleNameKey,
+        CNContactNamePrefixKey,
+        CNContactNameSuffixKey,
+        CNContactPreviousFamilyNameKey,
+        CNContactNicknameKey,
+        CNContactPhoneticGivenNameKey,
+        CNContactPhoneticMiddleNameKey,
+        CNContactPhoneticFamilyNameKey
+        ];
+
+    NSMutableArray *keys = [[NSMutableArray alloc]init];
+    [keys addObjectsFromArray:nameKeys];
+    [keys addObjectsFromArray:@[
+        CNContactEmailAddressesKey,
+        CNContactPhoneNumbersKey,
+        CNContactPostalAddressesKey,
+        CNContactOrganizationNameKey,
+        CNContactJobTitleKey,
+        CNContactImageDataAvailableKey,
+        CNContactNoteKey,
+        CNContactUrlAddressesKey,
+        CNContactBirthdayKey
+        ]];
+
+    if (withThumbnails) {
+        [keys addObject:CNContactThumbnailImageDataKey];
+    }
+
+    CNContactFetchRequest *fetchRequest = [[CNContactFetchRequest alloc] initWithKeysToFetch:keys];
+    [fetchRequest setUnifyResults:YES];
+
+    NSError *regExError = nil;
+    NSRegularExpression *wordRegex = [NSRegularExpression regularExpressionWithPattern:@"\\w+"
```

Comment:
you may want to check these errors and precreate them as part of the module init.

Reply:
So if I'm understanding this correctly, variable provided as the error argument in `NSRegularExpression` is filled with an error object should one get thrown during the operation. Are you suggesting that I make this regular expression in the init and return early if an error gets created?

Reply:
Yes, it is always the same, no reason to constantly create it.